### ISSUE TITLE
CORS Firebase Auth Cannot fetch to Auth url Capacitor 3

### ISSUE THREAD

**Issue by riderx** [firebase/firebase-js-sdk#5020]
### [REQUIRED] Describe your environment

  * Operating System version: IOS 14
  * Browser version: Capacitor 3
  * Firebase SDK version: json api
  * Firebase Product: auth

### [REQUIRED] Describe the problem

Since I wasn't able to use auth sdk in capacitor i tried to do it by json API using :
https://github.com/samuelgozi/firebase-auth-lite/blob/d4ed969c9d7a9181910b14f05a6e1140b33448d0/src/main.js#L118

This work perfectly on web but not in Mobile app because of the capacitor origin .

[Error] Failed to load resource: Origin capacitor://localhost is not allowed by Access-Control-Allow-Origin. (token, line 0)
https://securetoken.googleapis.com/v1/token?key=****

#### Steps to reproduce:
Create a starter capacitor app with the code below 

#### Relevant Code:

```javascript

const signUp = () => {
    const apiKey ="***"
    const url = `https://securetoken.googleapis.com/v1/token?key=${apiKey}`;
    return fetch(url, {
        method: 'POST',
        body: JSON.stringify({returnSecureToken: true})
    }).then(async response => {
        let data = await response.json();
        // If the response returned an error, try to get a Firebase error code/message.
        // Sometimes the error codes are joined with an explanation, we don't need that(its a bug).
        // So we remove the unnecessary part.
        if (!response.ok) {
            const code = data.error.message.replace(/ ?: [\w ,.'"()]+$/, '');
            throw Error(code);
        }
        return data;
    });
}
signUp()
```

**Comment by sam-gc** [comment]
Thanks for pointing this out. Since this is unrelated to the particular SDK version, I'm going to remove the `v9` label for now. I've filed an internal bug to track this, but unfortunately I can't make any predictions regarding timelines. (For Firebase folks: b/191185030)

**Comment by riderx** [comment]
@sam-gc thanks for the answer, I see in the doc you support Cordova so I supposed you support capacitor since all (most) modules are cross-compatible.
Thanks to filled a bug, with your answer in https://github.com/firebase/firebase-js-sdk/issues/5019 I should be able to use v9 thanks a lot!

**Comment by riderx** [comment]
Look like others got the issue : https://github.com/capacitor-community/firebase-analytics/issues/54

**Comment by riderx** [comment]
@sam-gc if you need the ionic team documented well the cors here : https://ionicframework.com/docs/troubleshooting/cors
**Capacitor**
Platform	Origin
iOS	capacitor://localhost
Android	http://localhost


**Ionic WebView 3.x plugin on Cordova**
Platform	Origin
iOS	ionic://localhost
Android	http://localhost


**Ionic WebView 2.x plugin on Cordova**
Platform	Origin
iOS	http://localhost:8080
Android	http://localhost:8080

**Comment by riderx** [comment]
@sam-gc please, my apps are broken because of that :/

**Comment by 1x2x3x4x** [comment]
Any news?

**Comment by bdhwan** [comment]
same issue. firebase 7.x works well. but 9.x not working because capacitor://

**Comment by simonpeters** [comment]
Anyone found a fix for this?

**Comment by m8xp0w3r** [comment]
Having the same issue. If anyones found a solution, please add it here. Thanks in advance.

**Comment by phpfs** [comment]
This fixed it for me: https://harryherskowitz.com/2021/08/23/firebase-capacitor.html

**Comment by riderx** [comment]
my solution was to remove Firebase and use Supabase instead, they care about people who use they tools

**Comment by wmadden** [comment]
It looks like the issue is Firebase incorrectly detecting the environment (see @alistairheath's [comment here](https://github.com/angular/angularfire/issues/3087#issuecomment-1063822400)).

There's a check in the `auth/compat` [library here](https://github.com/firebase/firebase-js-sdk/blob/4983f4d5a0dc385c5b3e042ace44c8204d3cce81/packages/auth-compat/src/platform.ts#L54) which checks for the URL scheme `"ionic://"`, which was used by older versions of Capacitor (it's now `"capacitor://"`) and if you set it back to `"ionic://"` (see the config option [`iosScheme`](https://capacitorjs.com/docs/config)) it seems to work again and doesn't attempt to load the gapi libraries.

TL;DR, in `capacitor.config` set `server: { iosScheme: "ionic" }`

**Comment by mesqueeb** [comment]
@wmadden can you please make a PR to fix this ?

**Comment by riderx** [comment]
@mesqueeb done here : https://github.com/firebase/firebase-js-sdk/pull/6236

**Comment by riderx** [comment]
The Pr is merged ! It will come soon in production !

**Comment by michaeljelly** [comment]
How soon is soon? (@mesqueeb) This is blocking an release/launch so it would be great to have a timeline.
Thanks so much for your work on fixing this up @riderx

**Comment by riderx** [comment]
@michaeljelly it has been released last week in the 9.8.2 👌

**Comment by michaeljelly** [comment]
Awesome, turns our I had a different problem where I wasn't initialising the app properly, which I have now fixed! So all good.
Thanks!
